security/rbac/engine/engine.go

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package engine
+
+import (
+	"github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+	"google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+)
+
+type AuthorizationArgs struct {
+	md 			metadata.MD
+	peerInfo 	*peer.Peer
+}
+
+// Decision is the enum type that represents different authorization
+// decisions a CEL engine can return.
+type Decision int32
+
+const (
+	// DECISION_ALLOW indicates allowing the RPC to go through.
+	DECISION_ALLOW Decision = iota
```

Comment:
Can we use camelcase for the variables defined here? Also I think the type Decision is a bit redundant, but it's ok to keep it.

Reply:
Yep! As a side note, the envoy RBAC names constants with all caps and underscores, but I'll change it to camelcase. Additionally, what do you mean by the type being redundant? If we need to return one of allow, deny, or unknown, I believe we do need a new enum to encapsulate these 3 values.

Reply:
Ah sorry, I forgot it. You are right:)